radis/lbl/factory.py

```diff
@@ -911,10 +911,14 @@ def eq_spectrum_gpu(
         molpar = MolParams()
 
         try:
-            id_set = self.df0[
-                "id"
-            ].unique()  # get all the molecules in the dataframe, should ideally be 1 element for GPU
-            mol_id = id_set[0]
+            if "id" in self.df0:
+                id_set = self.df0[
+                    "id"
+                ].unique()  # get all the molecules in the dataframe, should ideally be 1 element for GPU
+                mol_id = id_set[0]
```

Comment:
What if there's more than one element in id_set?

Reply:
The code before already seemed to assume one element. Just extended the same here.

Reply:
I would put an assertion here to be safe, otherwise if someone unknowingly extends to multiple id's in the future there will be no exception thrown here.